catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/stage/SceneTransitionTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.stage;
+
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.StartScript;
+import org.catrobat.catroid.content.bricks.ChangeVariableBrick;
+import org.catrobat.catroid.content.bricks.SceneStartBrick;
+import org.catrobat.catroid.content.bricks.SceneTransitionBrick;
+import org.catrobat.catroid.content.bricks.SetVariableBrick;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.formulaeditor.datacontainer.DataContainer;
+import org.catrobat.catroid.stage.StageActivity;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import static org.catrobat.catroid.uiespresso.util.UserVariableAssertions.assertUserVariableEqualsWithTimeout;
+import static org.catrobat.catroid.uiespresso.util.UserVariableAssertions.assertUserVariableNotEqualsForTimeMs;
+
+public class SceneTransitionTest {
```

Comment:
I'll look into this in the next few days.
I think https://github.com/Catrobat/Catroid/pull/3134 could be useful to test this.

Reply:
I added the whole test as CatrobatTestrunnerTests -if this is the prefered way I can remove this test.

Reply:
Ah, stupid me! I misunderstood that PR entirely. I didn't thought that projects have to be in `.catrobat`s. D'oh!

However, I still think that the _CatrobatTestrunner_ is suited here. It verifies that, when a scene switch brick is executed, the current scene stops (and doesn't execute any further bricks) and the new scene starts.

@84n4n4, just want to make sure that the _CatrobatTestrunner_ should be used here. And mostly that the _CatrobatTestrunner_ should be used in the future for such tests.
(I'll comment on this tests later)

Reply:
I checked with @84n4n4 and CatrobatTestrunner based tests are the way to go here, so I'll get rid of this espresso test.

Reply:
:kissing_cat:

Reply:
~(I'll comment on this tests later)~ → LGTM!

Reply:
One more thing: In my tests I assume unset variables to be equal to 0 and it works, but I suspect that could cause issues.
Should I use a scene for setting up variable values first in this case?

Reply:
> …but I suspect that could cause issues.

Are you worrying that the default value `0` may change in the future or do you have other issues in mind?

Reply:
I retested and my assumption was wrong (I thought that initial values could change based on the value of the variable during the last run). The tests are ok then 👍

Reply:
:+1:

> so I'll get rid of this espresso test

…, then move the ticket to _Ready For Code Review and I'll approve it :tada:

Reply:
~_then move the ticket to Ready For Code Review_~